### DISCUSSION TITLE
Found a space in the home path. We heavily discourage this, due to multiple     incompatibilities. Please check our docs for more information on this, as     well as some pointers on how to migrate to a different home path.

### DISCUSSION THREAD

**Discussion by homocomputeris** [pypa/pipx#1315] [Q&A] [answered]
I get the message every time I run pipx.

- What is home path here?
- Where are the docs on this issue? Nothing can be found here: https://www.google.com/search?hl=en&q=%22home%20path%22%20site%3Ahttps%3A%2F%2Fpipx.pypa.io%2Flatest%2F
- How to fix or disable this warning?

**Comment by dukecat0** [comment]
This should give you some information on it: https://pipx.pypa.io/stable/troubleshooting/#pipx-files-not-in-expected-locations-according-to-documentation

However, I think both the above docs and the warning message are not clear enough for normal users. Perhaps some improvements to them are needed.

**Reply by Gitznik** [reply]
Created an issue to improve the docs in this regard: https://github.com/pypa/pipx/issues/1320

**Comment by homocomputeris** [comment]
> This should give you some information on it

Ok, the problem is probably this:
```
pipx environment
⚠️ Found a space in the home path. We heavily discourage this, due to multiple incompatibilities.
    Please check our docs for more information on this, as well as some pointers on how to migrate
    to a different home path.
Environment variables (set by user):

PIPX_HOME=
PIPX_BIN_DIR=
PIPX_MAN_DIR=
PIPX_SHARED_LIBS=
PIPX_DEFAULT_PYTHON=
PIPX_FETCH_MISSING_PYTHON=
USE_EMOJI=

Derived values (computed by pipx):

PIPX_HOME=/Users/user/Library/Application Support/pipx
PIPX_BIN_DIR=/Users/user/.local/bin
PIPX_MAN_DIR=/Users/user/.local/share/man
PIPX_SHARED_LIBS=/Users/user/Library/Application Support/pipx/shared
PIPX_LOCAL_VENVS=/Users/user/Library/Application Support/pipx/venvs
PIPX_LOG_DIR=/Users/user/Library/Application Support/pipx/logs
PIPX_TRASH_DIR=/Users/user/Library/Application Support/pipx/.trash
PIPX_VENV_CACHEDIR=/Users/user/Library/Application Support/pipx/.cache
PIPX_STANDALONE_PYTHON_CACHEDIR=/Users/user/Library/Application Support/pipx/py
PIPX_DEFAULT_PYTHON=/usr/local/opt/python@3.12/libexec/bin/python
USE_EMOJI=true
```

Why does pipx sets `PIPX_HOME=/Users/user/Library/Application Support/pipx` while it's claimed to be `~/.local/pipx`?

**Reply by Gitznik** [reply, marked as answer]
It was the default location from version 1.3 to version 1.5. In 1.5 we moved it to `~/.local/pipx`. If the old one exists, we still use that one though, but print the warning you are seeing.

So if you installed Pipx between version 1.3 and 1.5, that's the behavior you see.

Please refer to this discussion on how to resolve it: https://github.com/pypa/pipx/discussions/1330

**Reply by peeeeter** [reply]
how can we move things to whatever will work bestfor mac os?  pipx reinstall-all didn't work, which is the suggestion on the docs.

**Reply by Gitznik** [reply]
Hey @peeeeter, you can find some pointers [here](https://pipx.pypa.io/stable/troubleshooting/#pipx-files-not-in-expected-locations-according-to-documentation).
Essentially you want to move all files under your pipx home (run `pipx environment` to figure out where that is) to `~/.local/pipx`. Then delete the folder of your old pipx home (if you copied, if you moved, then it's gone already), and then run `pipx reinstall-all`.

**Reply by gingerbeardman** [reply]
If pipx is responsible for the old and new locations, I would say it should also be responsible for migrating setups to avoid such a warning.

**Reply by chrysle** [reply]
We have already a script available to migrate from `pipsi` to `pipx`. I'd say a script to switch the installation to the recommended paths would be a welcome addition.
